CHANGELOG.md

```diff
@@ -1,6 +1,39 @@
 otj-metrics
 ===========
 
+4.0.13
+------
+* If running on Kubernetes the metric prefix becomes
+
+app_metrics.(applicationName).(environmentType).(environmentLocation).(clustername)-instance
+
+instead of 
+
+app_metrics.(applicationName).(environmentType).(environmentLocation).instance
+
+Example:
+
+app_metrics.service-demo.ci.sf.instance-1 becomes
+
+app_metrics.service-demo.ci.sf.arch-team-cluster-instance-1
+
+**This has been chosen as the least disruptive option but metrics relying
+on a certain count of announcements in an env-region might need fixing**
+
+- There's also an optional property
+
+`ot.graphite.reporting.include.cluster.type=separate`
+
+If you select this option:
+
+app_metrics.service-demo.ci.sf.instance-1 becomes app_metrics.service-demo.ci.sf.arch-team-cluster.instance-1
+
+This is superior in the future for metrics and dashboards, (separation of semantics) but requires rewriting your
+metrics and dashes.
+
+- In Singularity, the metrics prefix remains unchanged. 
```

Comment:
@mikebell90 sorry I had missed this before. Shouldn't it default to 'singularity' as cluster name instead of removing that path section? so that dashboards are consistent when a service is deployed to both k8s and mesos.

i.e.  app_metrics.service-demo.ci.sf.arch-team-cluster.instance-1 in k8s and  app_metrics.service-demo.ci.sf.singularity.instance-1

Reply:
However totally open to discussion here . At dentist so we’ll see if I make standup

Reply:
Yeah, I meant for separate=true option.

When I read "This is superior in the future for metrics and dashboards, (separation of semantics) but requires rewriting your metrics and dashes." I understood that the preferable approach is to have clusterName as a separate section and as I am creating new metrics (using AppMetrics) under completely new paths so I introduced clusterName as a seperate section without making it configurable.

Having different depths for the paths between k8s and mesos means we can't have shared dashboards either.